File: app/src/main/java/ru/roscha_akademii/medialib/videocontrol/view/VideoControlView.java
package ru.roscha_akademii.medialib.videocontrol.view;

import com.hannesdorfmann.mosby.mvp.MvpView;

public interface VideoControlView extends MvpView {
    void showTime(long duration);

    void showCurrentTime(long position);

    void showBuffered(long bufferedPosition);

    void setPlayPauseAction(PlayPauseMode mode);

    void setFullscreenAction(FullscreenMode mode);

    enum PlayPauseMode {
        PLAY, PAUSE;
    }

    public enum FullscreenMode {
        FULL, NORMAL
    }
}

File: app/src/main/java/ru/roscha_akademii/medialib/video/playvideo/videocontrol/view/VideoControlView.kt
package ru.roscha_akademii.medialib.video.playvideo.videocontrol.view

import com.hannesdorfmann.mosby.mvp.MvpView

interface VideoControlView : MvpView {
    fun showTime(duration: Long)

    fun showCurrentTime(position: Long)

    fun showBuffered(bufferedPosition: Long)

    fun setPlayPauseAction(mode: PlayPauseMode)

    fun setFullscreenAction(mode: FullscreenMode)

    enum class PlayPauseMode {
        PLAY, PAUSE
    }

    enum class FullscreenMode {
        FULL, NORMAL
    }
}
